readers/combined.cc

```diff
@@ -114,6 +116,10 @@ class mutation_fragment_merger {
                     }
                     return stop_iteration::no;
                 } else {
+                    if (_statistics && begin->fragment.is_clustering_row()) {
```

Comment:
I'd drop the check for is_clustering_row() and move the accounting before the `if`. After that change, we don't account for just clustering rows, but also static rows, partition starts, range tombstones, etc. It dilutes the meaning of "rows" in rows merged, but it's a more useful metric since it tells you how much data was merges vs not merged, regardless of exact type, which isn't very interesting.